### PR TITLE
rust-toolchain.toml: bump MSRV to 1.70.0

### PR DESCRIPTION
**Pull request by cole-h**
With Firefox 120 released on 21 November 2023, the MSRV is now 1.70.0.

---

According to https://firefox-source-docs.mozilla.org/writing-rust-code/update-policy.html (and after attempting to download Firefox myself), Firefox 120 has been released. Since helix follows Firefox's MSRV (https://github.com/helix-editor/helix/blob/47b6c4bc78d449e3586f28d26507bfc8fb8608e2/docs/CONTRIBUTING.md#minimum-stable-rust-version-msrv-policy), I bumped the MSRV to 1.70.0 in line with Firefox!

(The main thing I'm looking forward to is the sparse crates registry support added in 1.68.0.)

### CODE REVIEW

**Comment by cole-h** [comment]
I also wonder if you'd be amenable to adding the `rust-analyzer` component to the toolchain components list... I've had to add it to the list locally in the past to get a rust-analyzer working, but maybe that's a problem with my system configuration.

**Comment by pascalkuthe** [comment]
finally! We still want to wait till ubuntu updates but that is just a matter of time now.

I don't think we should include the rust-analyzer component, RA is backwards compatible (and using new version is preferable in every case, I always use nightly)

File: helix-term/src/commands/lsp.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ use std::{
 /// If there is no configured language server that supports the feature, this displays a status message.
 /// Using this macro in a context where the editor automatically queries the LSP
 /// (instead of when the user explicitly does so via a keybind like `gd`)
-/// will spam the "No configured language server supports <feature>" status message confusingly.
+/// will spam the "No configured language server supports \<feature>" status message confusingly.
```
Comment (cole-h) [inline helix-term/src/commands/lsp.rs:52]: I couldn't decide whether to backquote `<feature>` (so it appears as "code") or just escape the HTML tag rustdoc was complaining about. Happy to go either way.

**Review by the-mikedavis — COMMENTED** [review]
For CI you'll want to update the old references to `dtolnay/rust-toolchain@1.65` in `.github/workflows/*` to use `1.70` instead

**Comment by cole-h** [comment]
Good catch. I also notice that there's a `rust-version` in `helix-term/Cargo.toml` -- should I bump that to 1.70 as well? It's currently 1.65.

**Comment by pascalkuthe** [comment]
Yes

**Comment by archseer** [comment]
There's also a couple of TODOs in the code to simplify a couple things once MSRV is high enough, I think a couple can be tackled now

**Comment by cole-h** [comment]
Is that something you'd like me to handle in this PR, or do you want that to be a separate PR?

**Comment by pascalkuthe** [comment]
We usually handle that in the MSRV bump PR

**Comment by cole-h** [comment]
OK, I think I got everything in my latest commits.

I didn't know what you would prefer for the checked math error handling in 44efc0333432e16957a7e2a1687c2de05099ec25, so I just used  `expect` with the same message of `Selection ranges cannot overlap` (going off the comment above that block). I'll gladly implement a better approach, or even feel free to push directly to my branch if you don't want to deal with the back-and-forth.

**Comment by the-mikedavis** [comment]
Looks like ubuntu has bumped to 1.70 if I'm reading their packages correctly: https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=rust-all&searchon=names

File: helix-term/src/commands/lsp.rs (hunk 1 above)
Comment (the-mikedavis) [reply helix-term/src/commands/lsp.rs:52]: Escaping here seems proper to me 👍

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**